test/util/helpers.ts

```diff
@@ -225,22 +213,24 @@ export function stubProviderRequests(
       const stub = remainingStubs[stubIndex];
       const callNumber = callNumbersByRequest.get(stub.request) ?? 1;
 
-      cb(undefined, stub.response(request, callNumber));
+      const res = stub.response(callNumber);
 
       callNumbersByRequest.set(stub.request, callNumber + 1);
 
       if (!stub.remainAfterUse) {
         remainingStubs.splice(stubIndex, 1);
       }
+
+      return Promise.resolve(res);
```

Comment:
It looks like `stub.response` is a promise now. Do we want to `await` this? Then we don't need to wrap it in a promise below.

``` suggestion
      const result = await stub.response(callNumber);

      callNumbersByRequest.set(stub.request, callNumber + 1);

      if (!stub.remainAfterUse) {
        remainingStubs.splice(stubIndex, 1);
      }

      return result;